### PR TITLE
test: `walk_back` and `walk_range`

### PR DESCRIPTION
**Pull request by MegaRedHand**
This PR adds a test for the `ReverseWalker` created using `walk_back`. This also adds a missing case in the `walk_range` tests.

### CODE REVIEW

File: crates/storage/db/src/implementation/mdbx/cursor.rs [hunk 1]
```diff
@@ -102,6 +102,7 @@ impl<'tx, K: TransactionKind, T: Table> DbCursorRO<'tx, T> for Cursor<'tx, K, T>
         let start = match range.start_bound().cloned() {
             Bound::Included(key) => self.inner.set_range(key.encode().as_ref()),
             Bound::Excluded(key) => {
+                // unreachable!("Rust doesn't allow for Bound::Excluded in starting bounds");
                 self.inner
```
Comment (rkrasiuk) [inline crates/storage/db/src/implementation/mdbx/cursor.rs:106]:
```suggestion
            Bound::Excluded(key) => {
                self.inner
```

**Review by rkrasiuk — APPROVED** [review]

**Comment by rkrasiuk** [comment]
@MegaRedHand could u also add a check for invalid ranges (e.g. `3..1`, `3..=2`)

**Comment by MegaRedHand** [comment]
Sure! An empty range like `3..1` currently produces a walker with a single element (`3`). Should we change it so it creates an empty iterator? Or should we use an assert / return an error?

**Comment by gakonst** [comment]
Let's return an error in walk_range "if start bound less than end bound throw"

**Review by gakonst — APPROVED** [review]
LGTM - let's also add the invalid edges that Roman asked for

**Comment by MegaRedHand** [comment]
Added the check along with the tests, although I'm not sure what the error should be 🤔 . I left it as an `Error::Read(2)` for now, but I am open to suggestions.

**Review by gakonst — APPROVED** [review]
great!

**Pull request merged**